r/src/array_from_vector.cpp

```diff
@@ -192,6 +192,96 @@ std::shared_ptr<Array> MakeStructArray(SEXP df, const std::shared_ptr<DataType>&
   return std::make_shared<StructArray>(type, children[0]->length(), children);
 }
 
+std::shared_ptr<Array> MakeListArray(SEXP x, const std::shared_ptr<DataType>& type) {
```

Comment:
Largely copy-pasted from `MakeStringArray` which seems to have substantial overlap with the way to create a `ListArray` according to the layout described in the C++ manual, still not quite getting it right however

https://arrow.apache.org/docs/format/Columnar.html#format-columnar

Reply:
@romainfrancois could you please take a look at this and advise?

Reply:
I completely got rid of this, see the new recursive-builder method.